### ISSUE TITLE
implement rest of package

### ISSUE THREAD

**Issue by jordan-evens** [cffdrs/cffdrs_py#2]

**Comment by conbrad** [comment]
Any details on what's outstanding / priority? I'd be interested in contributing if you're looking for help.

**Comment by jordan-evens** [comment]
Hoping to implement FBP functions with names consistent with https://github.com/cffdrs/cffdrs_r/tree/jordan-evens/standardized-names - should just be a straight translation from R to python, at least initially. If you'd like to help that's awesome - thank you

**Comment by gagreene** [comment]
Hi there. I've written up a cffwis script that I'm happy to share (currently in a private repo). It accepts numpy arrays (for raster processing) and/or int/float values as inputs. I'm currently using it to calculate hourly and daily FWIS variables for ERA5-Land data for the entirety of Canada. I've done my best to use the variable naming conventions applied in the respective papers (e.g., Van Wagner 1977, 1987; etc.). Comparing my code with what you have, I'm noticing that I don't have the additional latitude adjustments for DMC and DC, but that would be simple enough to implement for numpy arrays. 

I have also developed a CFFBPS class that I can share (also currently in a private repo). It's fully functional for int/float inputs, but will have to be revised to accept numpy arrays. I have to flesh out the function and parameter data type descriptions, but otherwise it's ready to go. This class is part of a CanPyro module that I'm developing in collaboration with Dan Perrakis. It's structurally very different from the cffwis script (and your current fwi script) since it uses inheritance, but might be worth checking out?

I'm also happy to contribute to the cffdrs_py module if you need coding assistance. I regularly ask Justin Beckers for data, and have probably taken up a month of his time in the past few years. This is the least I can do to show my gratitude.

**Comment by gagreene** [comment]
I should also mention that I found an error in the fwi() function. I forked the repo a few weeks ago and applied the fix (just missing a negative sign in front of 0.023 on line 424). Happy to open a new issue if you'd prefer.

**Comment by jordan-evens** [comment]
The reference implementation for this is the R package, and this seems the same as that:
https://github.com/cffdrs/cffdrs_r/blob/main/R/fire_weather_index.r#L32
https://github.com/cffdrs/cffdrs_py/blob/main/cffdrs/fwi.py#L424

The paper at:
https://cfs.nrcan.gc.ca/pubwarehouse/pdfs/19973.pdf

has:
`1000 / (25 + 108.64 ( exp(-0.023 * BUI)`

But it's division in the R package and this, not multiplication. I think:
`x ^ (-n) == 1 / (x ^ n)`, so the sign is correct when it's using division?

**Comment by RobJago** [comment]
@jordan-evens @gagreene Yes, I have just a straight up non-object orientation using masked arrays (numpy.ma). It would be great to have QC on this code. I added in the -.023 fix in fwi() you recommended @gagreene. 

I don't have permissions to start a new repo here but I could add a file to the existing python repo. Let me know what works best.

**Comment by gagreene** [comment]
> The reference implementation for this is the R package, and this seems the same as that: https://github.com/cffdrs/cffdrs_r/blob/main/R/fire_weather_index.r#L32 https://github.com/cffdrs/cffdrs_py/blob/main/cffdrs/fwi.py#L424
> 
> The paper at: https://cfs.nrcan.gc.ca/pubwarehouse/pdfs/19973.pdf
> 
> has: `1000 / (25 + 108.64 ( exp(-0.023 * BUI)`
> 
> But it's division in the R package and this, not multiplication. I think: `x ^ (-n) == 1 / (x ^ n)`, so the sign is correct when it's using division?

Yeah, fair enough. I overlooked the divide in your equation. Please disregard my proposed edit!
